src/controllers/problem.ts

```diff
@@ -54,11 +56,25 @@ const problemController = {
             res.status(statusCodes.MISSING_PARAMS).json(errors.formatWith(errorMessage).array()[0]);
         } else {
             try {
+                // TODO: add verification that the fields within the Object problemMetadata are present
                 const problemData: IProblem = {
                     ...req.body,
-                    problemId: calculateProblemHash(req.body.problemId, req.body.source)
+                    problemId: calculateProblemHash(req.body.source.toUpperCase(), req.body.problemMetadata.platformProblemId.toUpperCase())
                 };
                 let newProblem: IProblemModel = await problemDBInteractions.create(new Problem(problemData));
+
+                if (newProblem.problemSetIds && newProblem.problemSetIds.length > 0) {
+                    for (const problemSetId of newProblem.problemSetIds) {
+                        let currProblemSet: IProblemSetModel = await problemSetDBInteractions.find(problemSetId);
+                        if (!currProblemSet.problemCount) {
+                            currProblemSet.problemCount = 1;
+                        } else {
+                            currProblemSet.problemCount += 1;
+                        }
+                        currProblemSet.save();
+                    }
+                }
```

Comment:
we should also add this logic to the update controller as well since a problemSet can be added to the problem after it's creation.